### PR TITLE
DOC: add steering council email to triage page + remove unactionable instructions

### PR DESCRIPTION
**Pull request by tacaswell**
## PR summary

Document how to nominate someone (or your self) to get triage rights.

### CODE REVIEW

**Comment by tacaswell** [comment]
Lets deal with that (good) problem when we have it.

File: doc/devel/triage.rst [hunk 1]
```diff
@@ -69,16 +69,14 @@ Triage team
...
+   recommend you to the triage team (look for "Member" on the top-right of
+   comments on GitHub).  If you worked with someone on the issues triaged, they
+   would be a good person to ask.
 3. Responsibly exercise your new power!
 
-Anyone with commit or triage rights may also nominate a user to be
-invited to join the triage team.
-
+Anyone with commit or triage rights may nominate a user to be invited to join
+the triage team by emailing matplotlib-steering-council@numfocus.org .
```
[the remark this answers is not in the document]
Comment (tacaswell) [reply doc/devel/triage.rst:79]: I think that is too detailed and prescriptive to put in the docs.

File: doc/devel/triage.rst [hunk 2]
```diff
@@ -18,8 +18,8 @@ are not part of the Matplotlib organization do not have `permissions
 to change milestones, add labels, or close issue
 <https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-permission-levels-for-an-organization>`_.
 If you do not have enough GitHub permissions do something (e.g. add a
-label, close an issue), please leave a comment tagging
-``@matplotlib/triageteam`` with your recommendations!
+label, close an issue), please leave a comment tagging with your
```
Comment (rcomer) [inline doc/devel/triage.rst:21]:
```suggestion
label, close an issue), please leave a comment with your
```

File: doc/devel/triage.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (tacaswell) [reply doc/devel/triage.rst:79]: I still do not think it is needed here.

[the remark this answers is not in the document]
Comment (rcomer) [reply doc/devel/triage.rst:79]: Whoever fields the email can always ask for more info if needed…

**Review by rcomer — APPROVED** [review]
This clearly fulfils the criteria "better than it was".

**Pull request merged**